src/components/TodoItem.tsx

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+// import { ErrorMessage } from '../App';
+import { ErrorMessage } from './Errors';
+
+interface TodoItemProps {
+  id: number;
+  title: string;
+  completed: boolean;
+  isLoading: boolean;
+  onToggle: (id: number) => void;
+  onDelete: (id: number) => void;
+  onError: (error: string) => void;
+}
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  id,
+  title,
+  completed,
+  isLoading,
+  onToggle,
+  onDelete,
+  onError,
+}) => {
+  const handleCheckboxChange = () => {
+    onToggle(id);
+  };
+
+  const handleDeleteTodo = () => {
+    if (id) {
+      onDelete(id);
+    } else {
+      onError(ErrorMessage.Delete);
+    }
+  };
+
+  return (
+    <div>
+      <div
+        data-cy="Todo"
+        className={`todo ${completed ? 'todo completed' : ''} ${isLoading ? 'is-active' : ''}`}
```

Comment:
Use the classnames library rules for add classes with condition, fix it everywhere